mmv1/products/dialogflowcx/TestCase.yaml

```
@@ -0,0 +1,451 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'TestCase'
+base_url: '{{parent}}/testCases'
+update_verb: :PATCH
+update_mask: true
+description: |
+  You can use the built-in test feature to uncover bugs and prevent regressions. A test execution verifies that agent responses have not changed for end-user inputs defined in the test case.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Official Documentation': 'https://cloud.google.com/dialogflow/cx/docs'
+  api: 'https://cloud.google.com/dialogflow/cx/docs/reference/rest/v3/projects.locations.agents.testCases'
+timeouts: !ruby/object:Api::Timeouts
+  insert_minutes: 40
+  update_minutes: 40
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  custom_import: templates/terraform/custom_import/dialogflowcx_test_case.go.erb
+  pre_create: templates/terraform/pre_create/dialogflow_set_location.go.erb
+  pre_update: templates/terraform/pre_create/dialogflow_set_location.go.erb
+  pre_delete: templates/terraform/pre_create/dialogflow_set_location.go.erb
+  pre_read: templates/terraform/pre_create/dialogflow_set_location.go.erb
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: 'dialogflowcx_test_case_full'
+    primary_resource_id: 'basic_test_case'
+    vars:
+      agent_name: 'dialogflowcx-agent'
+skip_sweeper: true
+id_format: '{{parent}}/testCases/{{name}}'
+import_format: ['{{parent}}/testCases/{{name}}']
```

If the only custom import logic is handling slashes in the parent name, it should be possible to use this import_format instead.

```suggestion
import_format: ['{{%parent}}/testCases/{{name}}']
```

---

TIL; in my defense, the other dialogflowcx resources use custom imports, probably because they predate `%parent` 😅.

TODO for myself: make a follow-up cleanup PR to update other dialogflowcx resources to use `%parent` assuming this works.